lib/nylas/model/attributable.rb

```diff
@@ -8,7 +8,11 @@ def self.included(model)
 
       def initialize(**initial_data)
         initial_data.each do |attribute_name, value|
-          send(:"#{attribute_name}=", value)
+          if attribute_definitions.key?(attribute_name)
```

Comment:
I'm wondering if rather than doing this check, would it make sense to make use of `method_missing`? I think that could potentially be more clear about the intentionality and be more resilient.

Reply:
in my experience, method missing is a bit harder to trace down what is going on. This is checking that an attribute is defined, and if so, calling the method. I could see the argument made that this violates "tell don't ask" in that we want to tell the model to assign the value, but the level of indirection that method missing/metaprogramming introduces doesn't feel like a valuable trade off to me.

Reply:
Yeah, it's a tradeoff either way... As long as this is definitely the only place that's calling these attributes in this way, then that's fine.

Reply:
In cases where someone calls `contact.field_that_doesnt_exist="foo"` it will continue to raise an `undefined method` error; which feels like it's in line with the principle of least surprise.